qiskit_nature/results/protein_folding_result.py

```diff
@@ -10,32 +10,68 @@
 # that they have been altered from the originals.
 """The protein folding result."""
-from typing import Union
+from __future__ import annotations
+import numpy as np
 
-from qiskit.opflow import PauliOp
 
-from qiskit_nature.problems.sampling.protein_folding.protein_folding_problem import (
-    ProteinFoldingProblem,
-)
+import qiskit_nature.problems.sampling.protein_folding.protein_folding_problem as pfp
+
 from qiskit_nature.results import EigenstateResult
+from .protein_folding_tools.protein_decoder import ProteinDecoder
+from .protein_folding_tools.protein_shape_file_gen import ProteinShapeFileGen
+from .protein_folding_tools.protein_plotter import ProteinPlotter
 
 
 class ProteinFoldingResult(EigenstateResult):
-    """The protein folding result."""
+    """
+    The Protein Folding Result.
+    """
 
     def __init__(
         self,
-        protein_folding_problem: ProteinFoldingProblem,
-        best_sequence: Union[str, PauliOp],
+        protein_folding_problem: pfp.ProteinFoldingProblem,
+        best_sequence: str,
     ) -> None:
+        """
+        Args:
+            protein_folding_problem: The protein folding problem that led to the result.
+            best_sequence: The best sequence in the result eigenstate.
+
+        """
         super().__init__()
         self._protein_folding_problem = protein_folding_problem
-        self._best_sequence: str = best_sequence
+        self._best_sequence = best_sequence
+        self._unused_qubits = self._protein_folding_problem.unused_qubits
+        self._main_chain_length = len(
+            self._protein_folding_problem.peptide.get_main_chain.main_chain_residue_sequence
+        )
+        self._side_chain_hot_vector = (
+            self._protein_folding_problem.peptide.get_side_chain_hot_vector()
+        )
+
+        self._protein_decoder = ProteinDecoder(
+            best_sequence=self._best_sequence,
+            side_chain_hot_vector=self._side_chain_hot_vector,
+            fifth_bit=5 in self._unused_qubits[:6],
+        )
+
+        self._protein_shape_file_gen = ProteinShapeFileGen(
+            self.protein_decoder.main_turns,
+            self.protein_decoder.side_turns,
+            self._protein_folding_problem.peptide,
+        )
+
+    @property
+    def protein_decoder(self) -> ProteinDecoder:
+        """Returns a ProteinDecoder.
+        This class will interpret the result bitstring and return the encoded information."""
+        return self._protein_decoder
 
     @property
-    def protein_folding_problem(self) -> ProteinFoldingProblem:
-        """Returns the protein folding problem."""
-        return self._protein_folding_problem
+    def protein_shape_file_gen(self) -> ProteinShapeFileGen:
+        """Returns a ProteinShapeFileGen. This class will take the encoded turns and
+        generate the position of every bead in the main and side chains."""
+        return self._protein_shape_file_gen
```

Comment:
I think we might want to limit docstring to what the method does, not the class.